lib/rouge/lexers/markdown.rb

```diff
@@ -102,7 +102,7 @@ def html
         end
 
         # links and images
-        rule %r/(!?\[)(#{edot}*?)(\])(?=[\[(])/ do
+        rule %r/(!?\[)(#{edot}*?|[^\]]*?)(\])(?=( ?\[|\())/ do
```

Comment:
White space between the link and its label is permitted in Gruber's Markdown implementation but is explicitly changed in [the CommonMark spec](https://spec.commonmark.org/0.29/#example-538) (see the discussion after example 539).

I'm a little torn on what to do. Gruber's implementation clearly permits it (and he expressly says it's permitted on his site) but the CommonMark spec authors make some good points as to why it's problematic.

What do you think?

Reply:
I totally missed that, and agree 100% with the CommonMark spec, as I don't want our writers to be creating links like that anyways. I'm going to pull it out, thanks!